### PR TITLE
Made plotScoreHeatmap more color blind friendly

### PR DESCRIPTION
**Pull request by dtm2451**
Zero exports or arguments changed in exported functions for colors adjustments.

Non-exported functions added to utils-heatmap.R for:

- creating discrete heatmap annotation colors
- creating max colors for continuous heatmap annotations
- parsing and picking colors for `annotation_col` and `annotation_row` dataframes
- for making annotation_colors input for pheatmap call, that is automatically generated based on `args$annotation_col` and `args$annotation_row` directly.
- note: These all get done directly before the pheatmap call, and only when the user did not supply their own `annotation_colors` to `...`

Docs updated to add pheatmap function to `@seealso` section

### CODE REVIEW

**Comment by dtm2451** [comment]
Future plans, though this would add an input:

- make labels easier to show as an annotation

File: R/utils.R [hunk 1]
```diff
@@ -21,3 +21,129 @@
...
+                
+                next.color.index.discrete <-
+                    next.color.index.discrete + length(in.this.annot)
+            } else {
+                # Make a 100 color split as in pheatmap code.
+                a <- cut(args$annotation_row[,i], breaks = 100)
+                # Assign to colors.
+                this.ramp <- annotation.colors.c[next.color.index.numeric]
+                new.colors <-
+                    grDevices::colorRampPalette(c("white",this.ramp))(100)[a]
```
Comment (dtm2451) [inline R/utils.R]: Note: grDevices / colorRampPalette are used by the pheatmap function internally, so this isn't really an extra dependency.

**Comment by dtm2451** [comment]
This isn't as ready as I'd thought late last night.  Will finish over the weekend, but:

- [x] Add `show.labels` and update docs.
- [ ] Ensure white is always assigned to the low end of continuous variables
- [ ] Make the rows and columns parts of `.make_heatmap_annotation_colors()` another internal function instead of a large code repeat.

File: R/utils-heatmap.R [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                next.color.index.discrete + length(in.this.annot)
+        } else {
+            # Make a 100 color split as in pheatmap code.
+            a <- cut(
+                annotation_df[order(annotation_df[,i]),i],
+                breaks = 100)
+            # Assign to colors.
+            this.ramp <- annotation.colors.n[next.color.index.numeric]
+            new.colors <-
+                grDevices::colorRampPalette(c("white",this.ramp))(100)[a]
```
Comment (dtm2451) [inline R/utils-heatmap.R]: grDevices / colorRampPalette are utilized by pheatmap, so this isn't really an extra depencdency.

Comment (LTLA) [reply R/utils-heatmap.R]: okay, but make sure you add it to `Suggests:`, if it isn't there already. This is actually a base package, so I'm not sure whether we even need to `Suggests:` is... let's see if CHECK complains.

Comment (dtm2451) [reply R/utils-heatmap.R]: BiocCheck didn't, but I haven't been able to run R CMD Check

**Pull request merged**